lib/include/openamp/rpmsg_virtio.h

```diff
@@ -22,9 +22,7 @@ extern "C" {
 #endif
 
 /* Configurable parameters */
-#ifndef RPMSG_BUFFER_SIZE
-#define RPMSG_BUFFER_SIZE	(512)
-#endif
+#define RPMSG_BUFFER_SIZE	@RPMSG_BUFFER_SIZE@
```

Comment:
Today the open-amp can be used without the cmakeand a library build, but with files directly build in the application project. 
This break the legacy , imposing the cmake usage (only optional version_def.h is handled by cmake).

And this would not guarantee that include files would be aligned  with a dynamic library. 

On the other side if the application projetc define the RPMSG_BUFFER_SIZE and open-amp library build as a sub project could this solve the issue?

I also wonder if a new APi to get the rpmsg _size such as the introduce in [Linux kernel ]()https://elixir.bootlin.com/linux/v5.16-rc2/source/include/linux/rpmsg.h#L189) would be more reliable.
The drawback would be for static allocation.

Reply:
I think making things more dependent on cmake is a possible problem. Also, how many people ever build with RPMSG_BUFFER_SIZE anything but 512?

Reply:
Well, I can only speak for myself, but I use OpenAMP with FreeRTOS, build with CMake and RPMSG_BUFFER_SIZE set to 4096.
What is so special about using RPMSG_BUFFER_SIZE to anything other than 512?

Reply:
From MPOV, it is also a valid use case to decrease the buffer size for MCU to MCU communications, to limit the shared memory.

I think we also have to think about evolution such as  https://github.com/OpenAMP/open-amp/pull/155 pull request is also a possible evolution that would make the size negociated.
That's why a new API could be nice to get the buffer effective size.
But as mentionned before this is not compatible with a static allocation. that said, that could allow to check the alignement between the library and the application...

Reply:
> Well, I can only speak for myself, but I use OpenAMP with FreeRTOS, build with CMake and RPMSG_BUFFER_SIZE set to 2048.
> What is so special about using RPMSG_BUFFER_SIZE to anything other than 512?

The only thing special about 512 is that Linux remoteproc is hard-coded to 512. I hadn't thought of the MCU->MCU case or other such configurations.

Reply:
@supergaute 

Could you confirm that https://github.com/OpenAMP/open-amp/pull/328 answer to your need and that we can close this one?

Reply:
Without any answer, I consider that the https://github.com/OpenAMP/open-amp/pull/328 answer to your need, please reopen this pull request if it is not the case